src/arrayMethodSort.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let operationCount = -1;
+
+    while (operationCount !== 0) {
+      operationCount = 0;
+
+      if (!compareFunction) {
+        for (let i = 0; i <= this.length - 1; i++) {
+          if (String(this[i]) > String(this[i + 1]) && this[i + 1]) {
+            [this[i], this[i + 1]] = [this[i + 1], this[i]];
+            operationCount++;
+          }
+        }
+      } else {
+        for (let i = 0; i <= this.length - 1; i++) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            [this[i], this[i + 1]] = [this[i + 1], this[i]];
+            operationCount++;
+          }
+        }
```

Comment:
set compare default function and void 2 loops

Reply:
done!